utils/src/main/scala/za/co/absa/enceladus/utils/types/TypedStructField.scala

```diff
@@ -178,6 +184,22 @@ object TypedStructField {
     }
   }
 
+  // BinaryTypeStructField
+  final class BinaryTypeStructField private[TypedStructField](structField: StructField)
+                                                             (implicit defaults: Defaults)
+    extends TypedStructFieldTagged[Array[Byte]](structField) {
+
+    // used to convert the default value from metadata's [[MetadataKeys.DefaultValue]]
+    override protected def convertString(string: String): Try[Array[Byte]] = {
+      //binary's default value is always expected to be in base64
+      Try(Base64.getDecoder.decode(string))
+    }
+
+    override def validate(): Seq[ValidationIssue] = {
+      ScalarFieldValidator.validate(this)
```

Comment:
Considering there is the `encoding` metadata, a dedicated validator is required.

Reply:
Great observation, I totally forgot about that...

So there is now a specific validator in place 😀